### DISCUSSION TITLE
Provide a more detailed documentation of the C++ API Reference.

### DISCUSSION THREAD

**Discussion by HuNingHe** [google-deepmind/mujoco#871] [Ideas]
I found that most of the tutorials are about python, and there are few examples about c++. And the documentation does not explain the API in detail. For example, the meaning of different indexes of "sensor_adr" and how to use it, I can only find a little use case of this in mujoco_mpc, which is not friendly to newcomers.

The most basic example should contain a controller that tells the user how to read the sensor data and how to set the joint torque.

**Comment by yuvaltassa** [comment]
I think you mean the [C API](https://mujoco.readthedocs.io/en/latest/APIreference/index.html)? There is a C++ API, but that is indeed not documented as it is not finalised.

The data structures are all documented [here](https://mujoco.readthedocs.io/en/latest/APIreference/APItypes.html#mjmodel) under struct types. For example 
```C
int*      sensor_adr;           // address in sensor array                  (nsensor x 1)
```

If this is still not clear enough, just browse the code, it is very easy to read. E.g. [here](https://github.com/deepmind/mujoco/blob/main/src/engine/engine_sensor.c)
